pkg/gadgets/socket-collector/tracer/bpf/tcp4-collector.c

```diff
@@ -56,7 +56,7 @@ static int dump_req_sock(struct seq_file *seq, struct tcp_request_sock *treq)
 
 	socket_bpf_seq_print(seq, proto, irsk->ir_loc_addr,
 		irsk->ir_num, irsk->ir_rmt_addr, irsk->ir_rmt_port,
-		TCP_SYN_RECV);
+		TCP_SYN_RECV, sock_i_ino(treq->req.req.sk));
```

Comment:
Are you sure for this one?
I let the `sock_i_ino()` because from my understanding `struct tcp_request_sock` has an inode but this [commit](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=10feb428a504) makes me doubting.

Reply:
The kernel example directly set 0, check [here](https://github.com/torvalds/linux/blob/eec4df26e24e978e49ccf9bcf49ca0f2ccdaeffe/tools/testing/selftests/bpf/progs/bpf_iter_tcp4.c#L188-L190).

Did you check what `ss` and `netstat` do?

Reply:
I did not check output of `netstat` because I did not find a reliable way to maintain a TCP socket in `TCP_SYN_RECV` state.

Reply:
Good point 🤔 ... I would follow what the kernel example does but maybe they were just trying to simplify the example.

Maybe @rata has any comment here?